### PR TITLE
Add rubocop

### LINKED ISSUE #99: Add Rubocop and move to shared config
* [ ] Grab a `.rubocop.yml` file from an existing @rubyatscale repo
* [ ] Fix violations
* [x] Update the CI action to use the [workflow defined in shared-config](https://github.com/rubyatscale/shared-config/blob/main/.github/workflows/ci.yml)

### PR DESCRIPTION
**Pull request by tstannard**
Add rubocop and fix violations 

"Adding the rubocop config from https://github.com/rubyatscale/code_teams/pull/14 to make sure that rubyatscale projects follow a consistent set of style rules." -Joey 

fixes #99

### CODE REVIEW

File: lib/code_ownership.rb [hunk 1]
```diff
@@ -95,10 +97,10 @@ def validate!(
     Private.load_configuration!
 
     tracked_file_subset = if files
-      files.select{|f| Private.file_tracked?(f)}
-    else
-      Private.tracked_files
-    end
+                            files.select { |f| Private.file_tracked?(f) }
```
Comment (ashleywillard) [inline lib/code_ownership.rb:100]: What's with all the white space?

File: lib/code_ownership/cli.rb [hunk 2]
```diff
@@ -53,16 +52,16 @@ def self.validate!(argv)
...
       parser.parse!(args)
 
       files = if options[:diff]
-        ENV.fetch('CODEOWNERS_GIT_STAGED_FILES') { `git diff --staged --name-only` }.split("\n").select do |file|
-          File.exist?(file)
-        end
-      else
-        nil
-      end
+                ENV.fetch('CODEOWNERS_GIT_STAGED_FILES') { `git diff --staged --name-only` }.split("\n").select do |file|
```
Comment (ashleywillard) [inline lib/code_ownership/cli.rb:59]: white space

File: lib/code_ownership/private/codeowners_file.rb [hunk 3]
```diff
@@ -64,22 +64,23 @@ def self.expected_contents_lines
             # 1) It allows the CODEOWNERS file to be used as a cache for validations
             # 2) It allows users to specifically see what their team will not be notified about.
             entry = if ignored_teams.include?(code_team.name)
-              "# /#{path} #{team_mapping}"
-            else
-              "/#{path} #{team_mapping}"
-            end
+                      "# /#{path} #{team_mapping}"
```
Comment (ashleywillard) [inline lib/code_ownership/private/codeowners_file.rb:67]: white space

**Review by ashleywillard — APPROVED** [review]

File: lib/code_ownership.rb (hunk 1 above)
Comment (tstannard) [reply lib/code_ownership.rb:100]: i know 🤖

File: lib/code_ownership/cli.rb (hunk 2 above)
Comment (tstannard) [reply lib/code_ownership/cli.rb:59]: 🏃‍♀️💨

File: lib/code_ownership/private/codeowners_file.rb (hunk 3 above)
Comment (tstannard) [reply lib/code_ownership/private/codeowners_file.rb:67]: 🚀

**Pull request merged**